ui/components/app/signature-request/signature-request.js

```diff
@@ -0,0 +1,355 @@
+
+import {
+  TextAlign,
+  TextColor,
+  TextVariant,
+  Size,
+  ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+  IconColor,
+  BackgroundColor,
+  Display,
+  BlockSize,
+  ///: END:ONLY_INCLUDE_IN
+} from '../../../helpers/constants/design-system';
+import {
+  BUTTON_VARIANT,
+  Button,
+  ButtonLink,
+  TagUrl,
+  Text,
+  ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+  Icon,
+  IconName,
+  ///: END:ONLY_INCLUDE_IN
+} from '../../component-library';
+
+///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+// eslint-disable-next-line import/order
+import Box from '../../ui/box/box';
+import { ENVIRONMENT_TYPE_NOTIFICATION } from '../../../../shared/constants/app';
+import { getEnvironmentType } from '../../../../app/scripts/lib/util';
+import { mmiActionsFactory } from '../../../store/institutional/institution-background';
+///: END:ONLY_INCLUDE_IN
+
+import Message from './signature-request-message';
+import Footer from './signature-request-footer';
+
+const SignatureRequest = ({ txData }) => {
+  const trackEvent = useContext(MetaMetricsContext);
+  const dispatch = useDispatch();
+  const t = useI18nContext();
+
+  const [hasScrolledMessage, setHasScrolledMessage] = useState(false);
+  const [showContractDetails, setShowContractDetails] = useState(false);
+  const [messageRootRef, setMessageRootRef] = useState(null);
+  const [messageIsScrollable, setMessageIsScrollable] = useState(false);
+
+  const {
+    id,
+    type,
+    msgParams: { from, data, origin, version },
+  } = txData;
+
+  // not forwarded to component
+  const hardwareWalletRequiresConnection = useSelector((state) =>
+    doesAddressRequireLedgerHidConnection(state, from),
+  );
+  const { chainId, rpcPrefs } = useSelector(getProviderConfig);
+  const unapprovedMessagesCount = useSelector(getTotalUnapprovedMessagesCount);
+  const subjectMetadata = useSelector(getSubjectMetadata);
+  const isLedgerWallet = useSelector((state) => isAddressLedger(state, from));
+  const { handleCancelAll } = useRejectTransactionModal();
+
+  ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+  // Used to show a warning if the signing account is not the selected account
+  // Largely relevant for contract wallet custodians
+  const selectedAccount = useSelector(getSelectedAccount);
+  const mmiActions = mmiActionsFactory();
+  const accountType = useSelector(getAccountType);
+  const isNotification = getEnvironmentType() === ENVIRONMENT_TYPE_NOTIFICATION;
+  const allAccounts = useSelector(accountsWithSendEtherInfoSelector);
+  const { address } = getAccountByAddress(allAccounts, from) || {};
+  const { custodySignFn } = useMMICustodySignMessage();
+  ///: END:ONLY_INCLUDE_IN
+
+  useEffect(() => {
+    setMessageIsScrollable(
+      messageRootRef?.scrollHeight > messageRootRef?.clientHeight,
+    );
+  }, [messageRootRef]);
+
+  const targetSubjectMetadata = txData.msgParams.origin
+    ? subjectMetadata?.[txData.msgParams.origin]
+    : null;
+
+  const parseMessage = () => {
```

Comment:
Before this was wrapped in lodash memoize. Should we continue to rely on memoize for this (for example using useMemo)? Or is there a reason why memoization is not needed anymore?

Reply:
The previous component had a lot of performance problems with larger payloads, the memoization was a way to alleviate that, since some OpenSea payloads were unusable.

We learned that the hard way: https://github.com/MetaMask/metamask-extension/pull/17052